### ISSUE TITLE
CognitoUserSession is not valid because idToken is null.

### ISSUE THREAD

**Issue by CaptainJack007** [aws-amplify/amplify-flutter#2230]
**i can't login and register new users.
when i want to login, it says:**
`W/CognitoUserSession( 9541): CognitoUserSession is not valid because idToken is null.
D/AWSMobileClient( 9541): Sending password.
D/AWSMobileClient( 9541): Using USER_SRP_AUTH for flow type.
W/AWSMobileClient( 9541): signalTokensNotAvailable
W/InternalCallback( 9541): Library attempted to call user callback twice, expected only once`

and my amplifyconfiguration.dart file :
`const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "aws2": {
                    "endpointType": "GraphQL",
                    "endpoint": "https://axsw6ccotvd5xivyrdm4hbyjnq.appsync-api.ap-northeast-2.amazonaws.com/graphql",
                    "region": "ap-northeast-2",
                    "authorizationType": "API_KEY",
                    "apiKey": "da2-wxoyznjwcngeho3pqcdlabpvgi"
                }
            }
        }
    },
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "AppSync": {
                    "Default": {
                        "ApiUrl": "https://axsw6ccotvd5xivyrdm4hbyjnq.appsync-api.ap-northeast-2.amazonaws.com/graphql",
                        "Region": "ap-northeast-2",
                        "AuthMode": "API_KEY",
                        "ApiKey": "da2-wxoyznjwcngeho3pqcdlabpvgi",
                        "ClientDatabasePrefix": "aws2_API_KEY"
                    },
                    "aws2_AWS_IAM": {
                        "ApiUrl": "https://axsw6ccotvd5xivyrdm4hbyjnq.appsync-api.ap-northeast-2.amazonaws.com/graphql",
                        "Region": "ap-northeast-2",
                        "AuthMode": "AWS_IAM",
                        "ClientDatabasePrefix": "aws2_AWS_IAM"
                    }
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": "ap-northeast-2:b8aeae1e-3cf9-4790-9738-2a45cf94acee",
                            "Region": "ap-northeast-2"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "ap-northeast-2_Nd0tH2uM4",
                        "AppClientId": "68d9f3em74j11p18rq86drf67t",
                        "Region": "ap-northeast-2"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH",
                        "socialProviders": [],
                        "usernameAttributes": [
                            "EMAIL"
                        ],
                        "signupAttributes": [
                            "EMAIL"
                        ],
                        "passwordProtectionSettings": {
                            "passwordPolicyMinLength": 8,
                            "passwordPolicyCharacters": []
                        },
                        "mfaConfiguration": "OFF",
                        "mfaTypes": [
                            "SMS"
                        ],
                        "verificationMechanisms": [
                            "EMAIL"
                        ]
                    }
                },
                "S3TransferUtility": {
                    "Default": {
                        "Bucket": "aws2-storage-2c0dc1a6171441-captaindev",
                        "Region": "ap-northeast-2"
                    }
                }
            }
        }
    },
    "storage": {
        "plugins": {
            "awsS3StoragePlugin": {
                "bucket": "aws2-storage-2c0dc1a6171441-captaindev",
                "region": "ap-northeast-2",
                "defaultAccessLevel": "guest"
            }
        }
    }
}''';`

**Comment by ragingsquirrel3** [comment]
Hey @CaptainJack007 sorry you are facing this issue. Do you mind telling us which versions of amplify-flutter plugins you are using? Is this happening on ios or android or both? Do you also mind including code blocks to show how you are calling Amplify auth methods in your code?

**Comment by Jordan-Nelson** [comment]
@CaptainJack007 - Can you please provide the info requested above? Thanks.

**Comment by Jordan-Nelson** [comment]
@CaptainJack007 - I am going to close this issue since we have not heard back and do not have enough info to attempt to reproduce this. If you are still experiencing this issue, please open a new issue and include the info in the bug report template. Thanks.
